src/Features/CSharp/Portable/PullMemberUp/CSharpMembersPullerService.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.  
+
+using System;
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeRefactorings.PullMemberUp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.CSharp.PullMemberUp
+{
+    [ExportLanguageService(typeof(IMembersPullerService), LanguageNames.CSharp), Shared]
+    internal class CSharpMembersPullerService : AbstractMembersPullerService<UsingDirectiveSyntax, CompilationUnitSyntax, NamespaceDeclarationSyntax>
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpMembersPullerService()
+        {
+        }
+
+        protected override UsingDirectiveSyntax GenerateNamespaceImportDeclaration(NamespaceDeclarationSyntax node, SyntaxGenerator generator)
+        {
+            return (UsingDirectiveSyntax)generator.NamespaceImportDeclaration(node.Name);
```

Comment:
this is not quite right.  consider something like:

```
namespace Outer {
  namespace Inner {
     class StartingPoint { ... }
  }
}
```

This will likely create `using Inner` instead of `using Outer.Inner`.  to resolve this, don't use syntax this way.  Instead, get the containing namespace symbol for the type you're in.  ANd then just generate the name for that and the corresponding namespace import.  this can all be done in teh base class.